lib/icinga/apiactions.cpp

```diff
@@ -555,5 +556,160 @@ Dictionary::Ptr ApiActions::GenerateTicket(const ConfigObject::Ptr&,
+		MacroProcessor::EscapeCallback(), nullptr, false
+	);
+
+	/* Check if resolved_command exists and it is of type command_type */
+	if (command_type == "CheckCommand") {
+		if (!CheckCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(404, "Can't find a valid " + command_type + " for '" + resolved_command + "'.");
+	} else if (command_type == "EventCommand") {
+		if (!EventCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(404, "Can't find a valid " + command_type + " for '" + resolved_command + "'.");
+	} else if (command_type == "NotificationCommand") {
+		if (!NotificationCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(404, "Can't find a valid " + command_type + " for '" + resolved_command + "'.");
+	}
+
+	/* Get TTL param */
+	if (!params->Contains("ttl"))
+		return ApiActions::CreateResult(400, "Parameter ttl is required.");
+
+	double ttl = HttpUtility::GetLastParameter(params, "ttl");
+	if (ttl <= 0)
+		return ApiActions::CreateResult(400, "Parameter ttl must be greater than 0.");
+
+	/* This generates a UUID */
+	String uuid = Utility::NewUniqueID();
+
+	/* Create the deadline */
+	double deadline = Utility::GetTime() + ttl;
+
+	/* Update executions */
+	Dictionary::Ptr pending_execution = new Dictionary();
+	pending_execution->Set("pending", true);
+	pending_execution->Set("deadline", deadline);
+	Dictionary::Ptr executions = checkable->GetExecutions();
+	if (!executions)
+		executions = new Dictionary();
+	executions->Set(uuid, pending_execution);
+	checkable->SetExecutions(executions);
+
+	/* Broadcast the update */
+	ApiListener::Ptr listener = ApiListener::GetInstance();
+	if (!listener)
+		return ApiActions::CreateResult(404, "No ApiListener instance available.");
+
+	Dictionary::Ptr updateParams = new Dictionary();
+	updateParams->Set("host", host->GetName());
+	if (service)
+		updateParams->Set("service", service->GetShortName());
+	updateParams->Set("executions", executions);
+
+	Dictionary::Ptr updateMessage = new Dictionary();
+	updateMessage->Set("jsonrpc", "2.0");
+	updateMessage->Set("method", "event::UpdateExecutions");
+	updateMessage->Set("params", updateParams);
+
+	MessageOrigin::Ptr origin = new MessageOrigin();
+	listener->RelayMessage(origin, checkable, updateMessage, true);
+
+	/* Create execution parameters */
+	Dictionary::Ptr execParams = new Dictionary();
+	execParams->Set("command_type", command_type);
+	execParams->Set("command", resolved_command);
+	execParams->Set("host", host->GetName());
+	if (service)
+		execParams->Set("service", service->GetShortName());
+
+	/*
+	 * If the host/service object specifies the 'check_timeout' attribute,
+	 * forward this to the remote endpoint to limit the command execution time.
+	 */
+	if (!checkable->GetCheckTimeout().IsEmpty())
+		execParams->Set("check_timeout", checkable->GetCheckTimeout());
+
+	execParams->Set("source", uuid);
+	execParams->Set("deadline", deadline);
```

Comment:
Everything is fine, but you forgot the `macros` parameter (as [here](https://github.com/Icinga/icinga2/blob/fd4abff369c7268d1c61219713b73c3c02c4b22f/lib/icinga/checkable-check.cpp#L555-L584)).